std/debug.zig

```diff
@@ -1053,7 +1053,7 @@ fn openSelfDebugInfoPosix(allocator: *mem.Allocator) !DwarfInfo {
     S.self_exe_file = try fs.openSelfExe();
     errdefer S.self_exe_file.close();
 
-    const self_exe_mmap_len = mem.alignForward(try S.self_exe_file.getEndPos(), mem.page_size);
+    const self_exe_mmap_len = mem.alignForward(@intCast(usize, try S.self_exe_file.getEndPos()), mem.page_size);
```

Comment:
I think this should be `std.math.cast` and then if it doesn't fit then return either `error.InvalidDebugInfo` or a new error code that means we can't support such a large file. As it stands, this invokes undefined behavior in release-fast & release-small builds for 32 bit targets if the executable file is larger than 4 GiB.

Reply:
cc @marler8997